### ISSUE TITLE
Bug in date picker when using dateInput/dateRangeInput with daylight savings time for Australia/Melbourne for "min" and "max" settings.

### ISSUE THREAD

**Issue by flickae** [rstudio/shiny#3103]
Hi Shiny Community,

I've got a very similar problem to #2204 where the date picker behaves oddly on the day Australia/Melbourne changes to/from daylight savings time. #2212 fixes problem #2204 where the value of the date picker cannot be set to the day that daylight savings changes. My issue is setting the "min" and "max" values of the date picker using dateInput() and dateRangeInput() as well as updateDateInput() and updateDateRangeInput().

**Reproducible Examples**
First set your system clock time zone to Australia/Melbourne (AEDT currently for daylight savings time or AEST for standard time). The date in question for these examples is 4th October, 2020 which is the date that Australia/Melbourne changed from standard time to daylight savings time.

**Case 1 - Blank End Date**
The case below leaves the end date blank as it can't set the value of "end" to be a "max" that doesn't exist.
```R
library(shiny)

ui <- fluidPage(
  dateRangeInput(inputId = "selectedDateRange", 
                             label = "",
                             start = "2020-01-19", 
                             end = "2020-10-04",
                             min = "2020-01-19",
                             max = "2020-10-04")
)

server <- function(input, output, session) { }

shinyApp(ui, server)
```

**Case 2 - Works Fine**
The case below works fine when the "max" date is not "2020-10-04".
```R
library(shiny)

ui <- fluidPage(
  dateRangeInput(inputId = "selectedDateRange", 
                             label = "",
                             start = "2020-01-19", 
                             end = "2020-10-04",
                             min = "2020-01-19",
                             max = "2020-10-05")
)

server <- function(input, output, session) { }

shinyApp(ui, server)
```

**Case 3 - Max Date Incorrect**
The case below will load "2020-10-03" as the end date, but in the date picker "2020-10-04" cannot be selected. The maximum that can be selected is "2020-10-03".
```R
library(shiny)

ui <- fluidPage(
  dateRangeInput(inputId = "selectedDateRange", 
                             label = "",
                             start = "2020-01-19", 
                             end = "2020-10-03",
                             min = "2020-01-19",
                             max = "2020-10-04")
)

server <- function(input, output, session) { }

shinyApp(ui, server)
```

**Case 4 - Min Date Incorrect**
When the same date is applied to "min", the start date can begin on "2020-10-04" correctly, however the minimum date allowed in the date picker is set to "2020-10-03".
```R
library(shiny)

ui <- fluidPage(
  dateRangeInput(inputId = "selectedDateRange", 
                             label = "",
                             start = "2020-10-04", 
                             end = "2020-10-19",
                             min = "2020-10-04",
                             max = "2020-10-19")
)

server <- function(input, output, session) { }

shinyApp(ui, server)
```

The same issue holds true for dateInput(), updateDateInput() and updateDateRangeInput(). Many thanks for the support.

**Comment by cpsievert** [comment]
Hi @flickae, I think this issue has been fixed in the latest version of shiny, would you mind confirming with `remotes::install_github("rstudio/shiny")` (and reopen the issue if it's still a problem)?

**Comment by dvg-p4** [comment]
This behavior is not actually fixed, can repo it all in rc-1.7.2 after setting my system clock to Melbourne.
